### PR TITLE
feat: GSTR3b detailed breackup report

### PR DESCRIPTION
**Pull request by ljain112**
Currently Only Section 4 (Eligible Itc) and 5 (Values of Exempt,Nil, Rated And Non-GST Supply) is implemented.

### CODE REVIEW

File: india_compliance/gst_india/report/gstr_3b_details/gstr_3b_details.py [hunk 1]
```diff
@@ -0,0 +1,368 @@
...
+        purchase_invoice = frappe.qb.DocType("Purchase Invoice")
+
+        query = (
+            frappe.qb.from_(purchase_invoice)
+            .select(
+                ConstantColumn("Purchase Invoice").as_("voucher_type"),
+                purchase_invoice.name.as_("voucher_no"),
+                purchase_invoice.posting_date,
+                purchase_invoice.eligibility_for_itc,
+                Sum(purchase_invoice.itc_integrated_tax).as_("integrated_tax"),
```
Comment (vorasmit) [inline india_compliance/gst_india/report/gstr_3b_details/gstr_3b_details.py:134]: check for is_opening

File: india_compliance/gst_india/report/gstr_3b_details/gstr_3b_details.py [hunk 2]
```diff
@@ -0,0 +1,368 @@
...
+                },
+            ]
+        )
+
+    def get_data(self):
+        formatted_data = []
+
+        invoices = self.get_inward_nil_exempt()
+
+        address_state_map = self.get_address_state_map()
```
Comment (vorasmit) [inline india_compliance/gst_india/report/gstr_3b_details/gstr_3b_details.py:284]: from document

File: india_compliance/gst_india/report/gstr_3b_details/gstr_3b_details.py [hunk 3]
```diff
@@ -0,0 +1,368 @@
...
+                address_state_map.get(invoice.supplier_address) or state
+            )
+            intra, inter = 0, 0
+            base_amount = invoice.base_amount
+
+            if (
+                invoice.is_nil_exempt == 1
+                or invoice.get("gst_category") == "Registered Composition"
+            ):
+                if supplier_state == place_of_supply:
```
Comment (vorasmit) [inline india_compliance/gst_india/report/gstr_3b_details/gstr_3b_details.py]: repeating code

File: india_compliance/gst_india/report/gstr_3b_details/gstr_3b_details.py [hunk 4]
```diff
@@ -0,0 +1,368 @@
...
+                {
+                    **invoice,
+                    "intra": intra,
+                    "inter": inter,
+                    "nature_of_supply": nature_of_supply,
+                }
+            )
+
+        self.data = sorted(
+            formatted_data, key=lambda k: (k["nature_of_supply"], k["posting_date"])
```
Comment (vorasmit) [inline india_compliance/gst_india/report/gstr_3b_details/gstr_3b_details.py:323]: .

**Review by vorasmit — CHANGES_REQUESTED** [review]

File: india_compliance/gst_india/report/gstr_3b_details/gstr_3b_details.py (hunk 2 above)
Comment (ljain112) [reply india_compliance/gst_india/report/gstr_3b_details/gstr_3b_details.py:284]: state is saved in display field , address template can change.

**Pull request merged**